src/codeview.jl

```diff
@@ -119,6 +119,7 @@ function cthulhu_typed(io::IO, debuginfo::Symbol,
     lineprinter = __debuginfo[debuginfo]
     rettype = ignorelimited(rt)
     lambda_io = IOContext(io, :limit=>true)
```

Comment:
Can we  not add the max_type_depth to this as discussed at https://github.com/JuliaDebug/Cthulhu.jl/issues/566#issuecomment-2077648580

Reply:
How would that be done?

Reply:
Oh sorry, I didn't realise you were passing maxtypedepth to mutliple print functions.
Anways, I would assume you could do `IOContext(io, :maxtypedepth=>CONFIG.type_depth_limit)` at the callsites where you are passing maxtypedepth or just redefine `io` to be that.

Reply:
Does `IOContext(io, :maxtypedepth=>CONFIG.type_depth_limit)` work? I don't see `maxtypedepth` in the docs:

```julia
help?> IOContext
search: IOContext

  IOContext

  IOContext provides a mechanism for passing output configuration settings among show methods.

  In short, it is an immutable dictionary that is a subclass of IO. It supports standard dictionary operations such as getindex, and can also be used as an I/O stream.

  ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  IOContext(io::IO, context::IOContext)

  Create an IOContext that wraps an alternate IO but inherits the properties of context.

  ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  IOContext(io::IO, KV::Pair...)

  Create an IOContext that wraps a given stream, adding the specified key=>value pairs to the properties of that stream (note that io can itself be an IOContext).

    •  use (key => value) in io to see if this particular combination is in the properties set

    •  use get(io, key, default) to retrieve the most recent value for a particular key

  The following properties are in common use:

    •  :compact: Boolean specifying that values should be printed more compactly, e.g. that numbers should be printed with fewer digits. This is set when printing array
       elements. :compact output should not contain line breaks.

    •  :limit: Boolean specifying that containers should be truncated, e.g. showing … in place of most elements.

    •  :displaysize: A Tuple{Int,Int} giving the size in rows and columns to use for text output. This can be used to override the display size for called functions, but to
       get the size of the screen use the displaysize function.

    •  :typeinfo: a Type characterizing the information already printed concerning the type of the object about to be displayed. This is mainly useful when displaying a
       collection of objects of the same type, so that redundant type information can be avoided (e.g. [Float16(0)] can be shown as "Float16[0.0]" instead of
       "Float16[Float16(0.0)]" : while displaying the elements of the array, the :typeinfo property will be set to Float16).

    •  :color: Boolean specifying whether ANSI color/escape codes are supported/expected. By default, this is determined by whether io is a compatible terminal and by any
       --color command-line flag when julia was launched.

  Examples
  ≡≡≡≡≡≡≡≡
```

Reply:
You can pass arbitrary keys, e.g. https://github.com/JuliaDebug/Cthulhu.jl/blob/17c53a157f5c21793115dc53ff83c87147dbc47b/src/codeview.jl#L143-146

Reply:
Interesting, sure I'll go ahead and pass `:maxtypedepth=>CONFIG.type_depth_limit`

Reply:
Let me know if I've not done that correctly

Reply:
Thanks, sorry but I think I forgot to explain why I wanted this which has probably caused some confusion.
Right now, you are passing maxtypedepth as the argument to lots of functions so that it's available in `show_annotation`.
The benefit to putting `maxtypedepth` in the io is that, the io is present in `show_annotation` so if you do `get(io, :maxtypedepth, nothing)` is `show_annotation` we can avoid passing `maxtypedepth` as an argument everywhere.
E.g.
```
io = IOContext(io, :a=>4)
f(io)

function f(io)
	g(io)
end

function g(io)
	a = get(io, :a, nothing)
	println(a)
end
```
Note that `a` didn't need to be passed into `f` or `g`. For another example, look how `vscode_io` is defined a couple lines below and how it is used in `show_annotation` to pass the `inlay_hints` and `diagnostics` variables.

Let me know if anything is unclear.

Reply:
Sounds good, I tried this out and I think it's working